traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -782,12 +782,17 @@ func getServers(h http.Header, params map[string]string, tx *sqlx.Tx, user *auth
 	usesMids := false
 	queryAddition := ""
 	dsHasRequiredCapabilities := false
+	var dsCDNName tc.CDNName
 	if dsIDStr, ok := params[`dsId`]; ok {
 		// don't allow query on ds outside user's tenant
 		dsID, err := strconv.Atoi(dsIDStr)
 		if err != nil {
 			return nil, 0, errors.New("dsId must be an integer"), nil, http.StatusNotFound, nil
 		}
+		_, dsCDNName, _, err = dbhelpers.GetDSNameAndCDNFromID(tx.Tx, dsID)
+		if err != nil {
+			return nil, 0, errors.New("ds not found"), nil, http.StatusNotFound, nil
+		}
```

Comment:
We should not need to check which CDN the delivery service is in. The query just needs to be adjusted to make sure the servers match the delivery service's CDN.

Reply:
well, it doesn't always check the cdn. for example:

GET /servers <-- doesn't check cdn
GET /servers?type=ORG <-- doesn't check cdn
GET /servers?dsId=X <-- does check cdn

Reply:
Right, but it always should check the CDN when the `dsId` query parameter exists, and all of the SQL for that part is in `deliveryServiceServersJoin` and `dssTopologiesJoinSubquery`.